### PR TITLE
NestJS file upload for user avatar-localStorage

### PR DESCRIPTION
**Pull request by IbrahimQumseya**

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -57,4 +82,26 @@ export class AuthController {
   ): Promise<User> {
     return this.authService.updateUser(user, updateUserDetailsDto);
   }
+
+  @Post('/upload')
```
Comment (sergiutoma) [inline src/auth/auth.controller.ts]: I think the route should be similar in name to the `@Get('/profile-image/:imageName')`
Create profile image route: `@POST('/upload/profile-image')`

File: src/auth/auth.controller.ts [hunk 2]
```diff
@@ -57,4 +82,26 @@ export class AuthController {
...
+  @ApiOkResponse({ description: 'Upload user Image' })
+  @ApiUnauthorizedResponse({ description: 'Invalid credentials' })
+  @UseGuards(AuthGuard('jwt'))
+  @UseInterceptors(FileInterceptor('file', storage))
+  uploadFile(@UploadedFile() file, @Request() req): Promise<User> {
+    const imagePath = file.filename;
+    const user: User = req.user;
+    console.log(user);
```
Comment (sergiutoma) [inline src/auth/auth.controller.ts]: You forgot a log 😉.

File: src/auth/users.respository.ts [hunk 3]
```diff
@@ -37,8 +40,35 @@ export class UsersRepository extends Repository<User> {
...
   async getUser(idUser: string): Promise<User> {
     const user = this.findOne({ where: { id: idUser } });
     return user;
   }
+
+  async updateOne(userId: string, imagePath: string): Promise<User> {
+    try {
+      const query = await this.createQueryBuilder()
+        .update(User)
+        .set({ profileImage: imagePath })
```
Comment (sergiutoma) [inline src/auth/users.respository.ts:54]: My IDE is showing me an error here:
getUserError
You forgot a log 😉.

**Comment by sergiutoma** [comment]
- These are routes to upload and download the user's profile picture, why did you choose to put in **auth.controller** and not in **user.controller**?
- When the new userProfile is uploaded, the old one must be deleted
- User profile images should not appear in git. Add content from **profileImages** to **gitIgnore**.
(You have to delete them all first and commit and then make the change to gitignore. The folder structure should appear(uploads/profileImages), only the images do not)

**Review by sergiutoma — CHANGES_REQUESTED** [review]
- These are routes to upload and download the user's profile picture, why did you choose to put in auth.controller and not in user.controller?

- When the new userProfile is uploaded, the old one must be deleted

- User profile images should not appear in git. Add content from profileImages to gitIgnore.

(You have to delete them all first and commit and then make the change to gitignore. The folder structure should appear(uploads/profileImages), only the images do not)
You worked fast, with these changes everything is perfect. Good job!!! 👏

File: src/auth/users.respository.ts (hunk 3 above)
Comment (IbrahimQumseya) [reply src/auth/users.respository.ts:54]: hmmm , in users entity does it has profileImage column ?

File: src/auth/users.respository.ts [hunk 4]
```diff
@@ -37,8 +41,35 @@ export class UsersRepository extends Repository<User> {
...
+        return user;
+      } else {
+        throw new NotFoundException();
+      }
+    } catch (error) {
+      throw new InternalServerErrorException();
+    }
+  }
+
+  async getProfileImage(user: User, res): Promise<Object> {
```
Comment (sergiutoma) [inline src/auth/users.respository.ts]: Should return <any> type
return-file
.

File: src/users/users.service.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+@Injectable()
+export class UsersService {
+  constructor(
+    @InjectRepository(UsersRepository)
+    private userRepository: UsersRepository,
+    @InjectRepository(UserDetailsRepository)
+    private userDetailsRepository: UserDetailsRepository,
+  ) {}
+
+  async getProfileImage(user: User, res): Promise<Object> {
```
Comment (sergiutoma) [inline src/users/users.service.ts]: Should return <any> type
return-file
.

**Review by sergiutoma — APPROVED** [review]
The code looks good! Good job! 👏 👍

File: src/users/users.service.ts (hunk 5 above)
Comment (IbrahimQumseya) [reply src/users/users.service.ts]: i did like this as u are saying , but its still the same problem
```
 const imageName = user.profileImage;
    const response: any = res.sendFile(
      join(process.cwd(), 'uploads/profileImages/' + imageName),
    );
```

**Pull request merged**